doc/index.md

```diff
@@ -119,27 +197,28 @@ Retrieves the next complete headers object (i.e. a block of headers or trailers)
 
 ### `stream:write_headers(headers, end_stream, timeout)` <!-- --> {#stream:write_headers}
 
-Write the given [`headers`](#http.headers) object to the stream.
-Takes a flag indicating if this is the last chunk in the stream, if `true` the stream will be closed.
+Write the given [`headers`](#http.headers) object to the stream. The function takes a flag indicating if this is the last chunk in the stream, if `true` the stream will be closed. If `timeout` is specified, the socket will be polled untile `timeout` is exceeded.
 
 
 ### `stream:get_next_chunk(timeout)` <!-- --> {#stream:get_next_chunk}
 
+(?) Does it block or poll? will it poll indefinetly or until some time?
+Returns the next buffered chunk of the http body, otherwise it blocks for input. The function will block until `timeout`, or indefinetly if no value is specified. If the message is compressed, runs inflate to decompress the data. On error, returns nil, an error message and an error number.
+
 
 ### `stream:unget(str)` <!-- --> {#stream:unget}
 
-Returns nothing
+Appends str to the front of the request body buffer. Returns nothing.
```

Comment:
"Prepends".

But this talks about the implementation, not what the outcome is:
Any :unget'd data will be returned from the next :get_next_chunk.